src/com/starkindustries/project/StarkDatabase.java

```diff
@@ -353,31 +353,37 @@ public boolean editComment(int c_id, String description, Connection conn) throws
 		else return true;
 	}
 	
-	public boolean voteQuestion(Question q, String type, Connection conn) throws SQLException {
+	public boolean voteQuestion(int q_id, String type, Connection conn) throws SQLException {
 		int rowsUpdated = 0;
 		Statement mystmt;
 		
-		int val = q.getTotal_votes();
+		/*int val = q.getTotal_votes();
 		if (type.equalsIgnoreCase("up")) val++;
-		else val--;
+		else val--;*/
+		char c;
+		if (type.equalsIgnoreCase("up")) c = '+';
+		else c = '-';
 		
-		String query = String.format("UPDATE questions SET total_votes = '%d' WHERE question_id = '%d'",val,q.getQuestion_id());
+		String query = String.format("UPDATE questions SET total_votes = total_votes %c 1 WHERE question_id = '%d'",c,q_id);
 		
 		mystmt = conn.createStatement();
 		rowsUpdated = mystmt.executeUpdate(query);
 		if (rowsUpdated == 0) return false;
 		else return true;
 	}
 	
-	public boolean voteAnswer(Answer a, String type, Connection conn) throws SQLException {
+	public boolean voteAnswer(int a_id, String type, Connection conn) throws SQLException {
```

Comment:
Could you revert voteAnswer to previous one? I have already connected the vote and your previous one is more straightforward imo.